.github/workflows/scan.yml

```diff
@@ -0,0 +1,39 @@
+name: puppet-lint scan
```

Comment:
This is an interesting thing. Do you have an example of where this ran so I can see the resulting report?

Reply:
This is just a sample (or sometimes called a starter workflow) that the user of the puppet-lint can add to their own repo, after we got all things done. User can use it to scan their repo and get code scanning alerts of the .pp files in their repo.
GitHub support this and the format used is SARIF format.

this is a run in my fork as example: (it is not intended to run against the puppet-lint repo itself, but the user's repo)
https://github.com/shaopeng-gh/puppet-lint/security/code-scanning

for people don't have permission:
![image](https://user-images.githubusercontent.com/81775155/164112572-1c6cb3e3-73eb-49b4-9a83-86fd6e3d1972.png)